azure_pipeline.yaml

```diff
@@ -13,7 +13,59 @@ variables:
   action:
 
 stages:
+  - stage: Precheck
+    jobs:
+      - job: Validate
+        pool:
+          vmImage: $(agentPool)
+        steps:
+        - task: AzureKeyVault@1
+          inputs:
+            ConnectedServiceName: azurerm-sandbox
+            keyVaultName: infra-vault-nonprod
+            secretsFilter: 'azure-devops-token'
+            runAsPreJob: false
+        - task: Bash@3
+          env:
+            thisbuild: $(Build.BuildId)
+            pipelinedefinition: $(System.DefinitionId)
+          inputs:
+            targetType: 'inline'
+            script: | 
+              #!/bin/bash
+              set -e
+
+              organization="hmcts"
+              project="CNP"
+
+              echo "This build number is: $thisbuild"
+
+              while [[ $thisbuild -ne $buildrunning ]]
+              do
+              JSON_DATA=$(curl -u :$(azure-devops-token) --request GET "https://dev.azure.com/$organization/$project/_apis/build/builds?api-version=5.1&definitions=$pipelinedefinition" -H "Content-Type: application/json" | jq  '.value[] | .status + (.id|tostring)' | sort -u)
+
+              for value in ${JSON_DATA[@]}; 
+              do 
+
+              number=(${value//[!0-9]/})
+              text=(${value//[0-9]/})
+              textcheck=$(sed -e 's/^"//' -e 's/"$//' <<<"$text")
+
+              if [[ "$textcheck" == "inProgress" ]];  then
```

Comment:
isn't it always sleeping for 30 seconds or am i misreading?

seems to be that it detects this build and waits then the while loop goes around and `thisbuild` is equal to `buildrunning`

Reply:
added if no builds inprogress to skip the while[]